test/property.test.js

```diff
@@ -51,36 +75,215 @@ describe('loopback:property generator', function() {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'year',
+      type: 'date',
+      defaultValue: '1466087191000'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('year');
+      expect(props.year).to.eql({
+        type: 'date',
+        default: '2016-06-16T14:26:31.000Z'
+      });
+      done();
+    });
+  });
+
   it('creates a typed array', function(done) {
     var propertyGenerator = givenPropertyGenerator();
     helpers.mockPrompt(propertyGenerator, {
       model: 'Car',
       name: 'list',
       type: 'array',
-      itemType: 'string'
+      customType: '', // temporary workaround for
+                      // https://github.com/yeoman/generator/issues/600
+      customItemType: '', // temporary workaround for
+                          // https://github.com/yeoman/generator/issues/600
+      itemType: 'object'
     });
 
     propertyGenerator.run(function() {
       var definition = common.readJsonSync('common/models/car.json');
       var prop = definition.properties.list;
-      expect(prop.type).to.eql(['string']);
+      expect(prop.type).to.eql(['object']);
+      done();
+    });
+  });
+
+  it('creates string item typed array', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'options',
+      type: 'array',
+      itemType: 'string',
+      defaultValue: 'AWD,3.2L, navigation'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('options');
+      expect(props.options).to.eql({
+        type: ['string'],
+        default: ['AWD','3.2L','navigation']
+      });
+      done();
+    });
+  });
+
+  it('creates number item typed array', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'parts',
+      type: 'array',
+      itemType: 'number',
+      defaultValue: '123456, 98765'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('parts');
+      expect(props.parts).to.eql({
+        type: ['number'],
+        default: [123456, 98765]
+      });
+      done();
+    });
+  });
+
+  it('creates boolean item typed array', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'certifications',
+      type: 'array',
+      itemType: 'boolean',
+      defaultValue: 't,t'
```

Comment:
Even better, list all supported values (`t`, `true`, `f`, `false`, etc.) to make sure all of them are covered by a test.